hare/broker.go

```diff
@@ -156,34 +169,72 @@ func (b *Broker) eventLoop() {
 	}
 }
 
+func (b *Broker) updateCurrentLayer(id InstanceId) {
+	if id > b.currLayer { // update only newer layers
+		b.currLayer = id
+	}
+}
+
+func (b *Broker) updateSynchronicity(id InstanceId) {
+	if _, ok := b.outbox[id]; ok { // already has result
+		return
+	}
+
+	// not exist means unknown, check & set
+
+	if !b.isNodeSynced() {
+		b.outbox[id] = nil // mark invalid
```

Comment:
I don't like the way you use `outbox` to store the sync state. It's not readable, please use a dedicated map.

Reply:
Changed it. Please review again.